src/hidra/receiver/plugins/asapo_producer.py

```diff
@@ -230,9 +229,21 @@ def __init__(self, endpoint, beamtime, token, n_threads, file_regex,
         self.n_threads = n_threads
         self.timeout = timeout
         self.default_data_source = default_data_source
-        self.file_regex = file_regex
         self.start_file_idx = start_file_idx
 
+        # Set and validate regexp
+        expected_keys = ["scan_id", "file_idx_in_scan"]
+        if self.default_data_source is None:
+            expected_keys.append("data_source")
+        try:
+            self.file_regex = re.compile(file_regex)
+        except Exception as e:
+            raise utils.UsageError("Compilation if regexp %s failed", file_regex)
+
+        for expected_key in expected_keys:
+            if expected_key not in self.file_regex.groupindex:
+                raise utils.UsageError("Expected regexp group %s is not in regexp", expected_key)
```

Comment:
The Python documentation of the `re` module uses "regex" but not `regexp`
```suggestion
                raise utils.UsageError("Expected regex group %s is not in regex", expected_key)
```